tests/tracing/_regtest_outputs/test_dnatracing.test_tracestats.out

```diff
@@ -1,24 +1,24 @@
                  Contour Lengths  Circular  End to End Distance Image Name Basename
-Molecule Number                                                                    
-0                     225.487215     False            35.015017        tmp        .
+Molecule Number
+0                     225.607081     False            35.015017        tmp        .
 1                      45.824632      True             0.000000        tmp        .
-2                     110.326409     False            21.881598        tmp        .
-3                      94.436535      True             0.000000        tmp        .
+2                     127.758534     False            22.092483        tmp        .
```

Comment:
I have been looking into this but am struggling to find a good way of visualising it. @ns-rse do you have any suggestions as to how I should investigate this change? I tried visualising it by running topostats with the parameters used in the test on both my branch and dev, however I get a problem where my branch doesn't pick up on one of the grains, indicating that there is something amiss in my configuration when trying to imitate the tests. 

Perhaps there is a way to get the regression test to save an image of the grain masks during the test?

Reply:
A common plot that I use when comparing two methods is [Bland-Altman plot](https://en.wikipedia.org/wiki/Bland–Altman_plot) and so you could for each metric plot the "old" v "new" values and see if there are any extreme values.

That doesn't tell you what has changed though and the only way I can think of doing that is to compare current `dev` output (i.e. before any change to flattening) to this branch. Grains are numbered from top-left to bottom right if I remember correctly (or perhaps the reverse :thinking: ) and what would really be required is to extract plots of the skeletonised molecules and compare them as my understanding is that these statistics are based on the post-skeletonisation traces.

Not sure if that is helpful (pretty sure the Bland-Altman stuff isn't immediately relevant but worth bearing in mind when comparing analytical methods in general).

Reply:
I have created a difference comparison of the binary grain masks that are passed to the flattening functions, I see only minor changes:

![diff](https://user-images.githubusercontent.com/86117496/201328296-521660b2-7bd3-4109-aac4-926933ea1d24.png)

Reply:
@SylviaWhittle So I've been looking into this and this is what I've found:
- The following plots show the differences between the image in the region around the second grain when processing with otsu. We see not much change:
<img width="1054" alt="Screenshot 2022-11-11 at 14 51 22" src="https://user-images.githubusercontent.com/91465918/201365019-db7015eb-02ff-4fb0-9178-529a4d8a0329.png">

- This is reflected in the mask and difference masks:
<img width="1053" alt="Screenshot 2022-11-11 at 14 51 49" src="https://user-images.githubusercontent.com/91465918/201365117-b2a44fb4-e392-4c88-8494-5883e60a1e24.png">
<img width="458" alt="Screenshot 2022-11-11 at 14 52 05" src="https://user-images.githubusercontent.com/91465918/201365173-1a997450-ec7f-43a5-9672-79b3172cd15b.png">

- When we look at the traces that DNA tracing has made, we find that the disordered traces differ, the ordered traces become more aligned and the splined traces are not generated from the fitted trace. This means the difference in the fitted trace is causing the issue, and we can see it's down to the janky part of the fitted trace getting larger:
<img width="925" alt="Screenshot 2022-11-11 at 14 59 44" src="https://user-images.githubusercontent.com/91465918/201366678-45cd8c06-c6fa-4058-ac23-01a6167bbed4.png">

- The slight increase in pixel heights around this area in the dev branch may be the reason for this:
<img width="505" alt="Screenshot 2022-11-11 at 15 03 27" src="https://user-images.githubusercontent.com/91465918/201367351-031753e6-2d83-40d2-9c1e-32a4f95b7c38.png">

Therefor this might be one to tally up to poor molecule tracing and not a change of behaviour in flattening.

Reply:
Thank you very much for this in-depth analysis @MaxGamill-Sheffield. I concur. The flattening is important to get right, and through histograms, experimentalist opinion and median plotting I am able to show that the new flattening is better, so I would say this could be left up to tweaks in the tracing functions to address.

Resolving the conversation, un-resolve if you don't agree 😄